darshan-util/pydarshan/darshan/tests/test_plot_dxt_heatmap.py

```diff
@@ -115,21 +93,19 @@ def test_set_x_axis_ticks_and_labels(
     if filepath is None:
         # don't have any data sets with a max time between 1 and 10, so
         # create a synthetic one here
-        data = [[4, 1.03378843, 1.03387713, 0], [4000, 1.04216653, 1.04231459, 0]]
-        cols = ["length", "start_time", "end_time", "rank"]
-        agg_df = pd.DataFrame(data=data, columns=cols)
+        runtime = 1
 
     else:
         filepath = get_log_path(filepath)
         # for all other data sets just load the data from the log file
+        # calculate the elapsed runtime
         report = darshan.DarshanReport(filepath)
-        agg_df = heatmap_handling.get_aggregate_data(
-            report=report, mod="DXT_POSIX", ops=["read", "write"]
-        )
+        runtime = report.metadata["job"]["end_time"] - report.metadata["job"]["start_time"]
+        runtime = max(runtime, 1)
```

Comment:
How does this help when `runtime` was rounded down from `1.3` seconds? (i.e., is the approach used in the test the sustainable approach?). It feels like it is just kind of a fragment of the true logic, and we're actually using `tmax` for this test, but not in general? It seems confusing.

Reply:
Ah, I guess the `tmax` is just for setting the ticks, though my comment about thoroughly handling the desired runtime compared to what happens in source remains here, I think.

Reply:
Yeah that's a good point. I'll add in the full logic.